src/paint/map_element/fence.c

```diff
@@ -14,10 +14,384 @@
+        if (ps != NULL) {
+            ps->tertiary_colour = dword_141F71C;
+        }
+    } else {
+        paint_struct * ps;
+
+        ps = sub_98197C(imageId, offset.x, offset.y, boundsL1.x, boundsL1.y, boundsL1.z, offset.z, boundsL1_.x, boundsL1_.y, boundsL1_.z, get_current_rotation());
+        if (ps != NULL) {
+            ps->tertiary_colour = dword_141F71C;
+        }
+
+        ps = sub_98199C(imageId + 1, offset.x, offset.y, boundsL1.x, boundsL1.y, boundsL1.z, offset.z, boundsL1_.x, boundsL1_.y, boundsL1_.z, get_current_rotation());
+        if (ps != NULL) {
+            ps->tertiary_colour = dword_141F71C;
+        }
+    }
 }
+
+void do_part_2(uint32 dword_141F708, const rct_scenery_entry * dword_141F70C, uint32 dword_141F710, uint32 dword_141F714, uint32 dword_141F718, uint32 dword_141F71C, uint32 imageOffset, rct_xyz16 offset, rct_xyz16 bounds, rct_xyz16 boundsOffset)
+{
+    uint32 baseImageId = dword_141F70C->image + imageOffset + dword_141F708;
+    uint32 imageId = baseImageId;
+
+
+    if (dword_141F70C->wall.flags & WALL_SCENERY_FLAG2) {
+        if (dword_141F70C->wall.flags & WALL_SCENERY_FLAG1) {
+            imageId |= dword_141F714;
+        }
+
+        if (dword_141F710 != 0) {
+            imageId = (imageId & 0x7FFFF) | dword_141F710;
+        }
+
+        sub_98197C(imageId, offset.x, offset.y, bounds.x, bounds.y, bounds.z, offset.z, boundsOffset.x, boundsOffset.y, boundsOffset.z, get_current_rotation());
+        if (dword_141F710 == 0) {
+            imageId = baseImageId + dword_141F718;
+            sub_98199C(imageId, offset.x, offset.y, bounds.x, bounds.y, bounds.z, offset.z, boundsOffset.x, boundsOffset.y, boundsOffset.z, get_current_rotation());
+        }
+    } else {
+        if (dword_141F70C->wall.flags & WALL_SCENERY_FLAG1) {
+            imageId |= dword_141F714;
+        }
+
+        if (dword_141F710 != 0) {
+            imageId = (imageId & 0x7FFFF) | dword_141F710;
+        }
+
+        paint_struct * paint = sub_98197C(imageId, offset.x, offset.y, bounds.x, bounds.y, bounds.z, offset.z, boundsOffset.x, boundsOffset.y, boundsOffset.z, get_current_rotation());
+        if (paint != NULL) {
+            paint->tertiary_colour = dword_141F71C;
+        }
+    }
+}
+/**
+ * rct2: 0x006E44B0
+ * @param direction (cl)
+ * @param height (dx)
+ * @param map_element (esi)
+ */
+void fence_paint(uint8 direction, int height, rct_map_element * map_element)
+{
+    rct_drawpixelinfo * dpi = RCT2_GLOBAL(0x140E9A8, rct_drawpixelinfo *);
+    RCT2_GLOBAL(RCT2_ADDRESS_PAINT_SETUP_CURRENT_TYPE, uint8) = VIEWPORT_INTERACTION_ITEM_WALL;
+
+    rct_scenery_entry * dword_141F70C = g_wallSceneryEntries[map_element->properties.fence.type];
+    uint32 dword_141F708 = 0;
+
+    if (dword_141F70C->wall.flags2 & WALL_SCENERY_FLAG5) {
+        // animation?
+        dword_141F708 = (gCurrentTicks & 7) * 2;
+    }
+
+
+    uint32 dword_141F714 = (map_element->properties.fence.item[1] & 0b11111) << 19 | 0x20000000;
```

Comment:
`primaryColour = map_element->properties.fence.item[1] & 0x1F;` Then build it up from that